SWI-cpp2.h

```diff
@@ -583,6 +582,34 @@ class PlTerm_recorded : public PlTerm
   explicit PlTerm_recorded(record_t r) { PlCheck(PL_recorded(r, C_)); }
 };
 
+class PlModule : public WrappedC<module_t>
+{
+public:
+  explicit PlModule(const std::string& name)
+    : WrappedC<module_t>(PL_new_module(PlAtom(name).C_))
+  { verify();
+  }
+  explicit PlModule(PlAtom name)
+    : WrappedC<module_t>(PL_new_module(name.C_))
+  { verify();
+  }
+};
+
+class PlPredicate : public WrappedC<predicate_t>
+{
+public:
+  explicit PlPredicate()
+    : WrappedC<predicate_t>(PlPredicate::null) { }
```

Comment:
I don't think there is any scenario I'm aware of where it is useful to have a NULL predicate pointer.

Reply:
Removed (it was in rocksdb, but I've now made it explicit)